pkg/k8s/watchers/pod.go

```diff
@@ -475,6 +485,14 @@ func (k *K8sWatcher) deleteK8sPodV1(pod *slim_corev1.Pod) error {
 		k.redirectPolicyManager.OnDeletePod(pod)
 	}
 
+	podNSName := k8sUtils.GetObjNamespaceName(&pod.ObjectMeta)
+	podEP := k.endpointManager.LookupPodName(podNSName)
+	if podEP != nil && podEP.NoTrackPort != "" {
+		podEP.UpdateNoTrackRules(func(ns, podName string) (noTrackPort string, err error) {
```

Comment:
Can we guarantee that it's appropriate to handle this logic through an eventqueue as is done here? I'm thinking that if `Endpoint.Delete()` is called first, then this event will be cancelled before the logic is executed, and cleanup of the iptables rules will not occur.

What are the ordering guarantees between the issue of the CNI delete from kubelet vs. the receipt & processing of the pod delete event from the apiserver? 

If we can guarantee that the pod delete will be processed (and we return from handling the event here) before the CNI delete is issued by kubelet, then this won't be a problem but if we can't get a strong guarantee, then it might be better to just run the underlying cleanup logic directly from either `Endpoint.Delete()` or `Endpoint.leaveLocked()`.

Reply:
Good point, I don't think there's any strong guarantee which API calls hit the agent first, moved the logic to `Endpoint.Delete()`.